src/clawback_escrow/src/test/clawback.test.ts

```diff
@@ -0,0 +1,171 @@
+  );
+
+  const contractMetadata = await pxes[0].getContractMetadata(escrowInstance.address);
+  expect(contractMetadata).toBeDefined();
+  expect(contractMetadata.isContractPubliclyDeployed).toBeFalsy();
+
+  const escrowContract = await escrowDeployment.send().deployed();
+  logger.info('escrow deployed', escrowContract.address);
+  return escrowContract;
+}
+
+async function deployClawbackEscrow(pxes: PXE[], wallet: Wallet) {
+  const clawbackSecretKey = Fr.random();
+  const clawbackPublicKeys = (await deriveKeys(clawbackSecretKey)).publicKeys;
+  const clawbackDeployment = ClawbackEscrowContract.deployWithPublicKeys(clawbackPublicKeys, wallet);
+  const clawbackInstance = await clawbackDeployment.getInstance();
+
+  await Promise.all(
+    pxes.map(async (pxe) => pxe.registerAccount(clawbackSecretKey, await computePartialAddress(clawbackInstance))),
+  );
+
+  const clawbackContract = await clawbackDeployment.send().deployed();
+
+  logger.info(`clawback address: ${clawbackContract.address}`);
+
+  return clawbackContract;
+}
+
+describe.only('Clawback Escrow - Multi PXE', () => {
+  let alicePXE: PXE;
+  let bobPXE: PXE;
+
+  let aliceWallet: AccountWalletWithSecretKey;
+  let bobWallet: AccountWalletWithSecretKey;
+
+  let alice: AccountWallet;
+  let bob: AccountWallet;
+
+  let token: TokenContract;
+  let escrow: EscrowContract;
+  let clawback: ClawbackEscrowContract;
+
+  let logger: Logger;
+
+  beforeAll(async () => {
+    alicePXE = await createPXE(0);
+    bobPXE = await createPXE(0);
+
+    aliceWallet = await createAccount(alicePXE);
+    bobWallet = await createAccount(bobPXE);
+
+    alice = aliceWallet;
+    bob = bobWallet;
+
+    console.log({
+      alice: alice.getAddress(),
+      bob: bob.getAddress(),
+    });
+  });
+
+  beforeEach(async () => {
+    token = (await deployToken(alice)) as TokenContract;
+    clawback = (await deployClawbackEscrow([alicePXE, bobPXE], alice)) as ClawbackEscrowContract;
+    // escrow = await deployEscrow([alicePXE, bobPXE], alice, clawback.address) as EscrowContract;
+    escrow = (await deployEscrow([alicePXE, bobPXE], alice, clawback.address)) as EscrowContract;
+    console.log({
+      token: token.address,
+      clawback: clawback.address,
+      escrow: escrow.address,
+    });
+    // alice and bob know the token contract
+    await alicePXE.registerContract(token);
+    await bobPXE.registerContract(token);
+
+    await alicePXE.registerContract(escrow);
+    await bobPXE.registerContract(escrow);
+
+    await alicePXE.registerContract(clawback);
+    await bobPXE.registerContract(clawback);
+
+    await alicePXE.registerSender(clawback.address);
+    await alicePXE.registerSender(escrow.address);
+
+    await bobPXE.registerSender(clawback.address);
+    await bobPXE.registerSender(alice.getAddress());
+
+    bob.setScopes([bob.getAddress(), alice.getAddress(), escrow.address, clawback.address]);
```

Comment:
can u try if this works?
```suggestion
    bob.setScopes([bob.getAddress(), escrow.address]);
```

Reply:
Yeah. I still have reviewing the whole account setup (in all e2e). I think there was some error (or confusion on my part) in the note discovery that made me always overreach on the scope/permissions.
Your suggestion works :)

Reply:
but bob shouldn't have alice's keys to be able to set her address as scope